### ISSUE TITLE
Update admin password on install from tasks.py when installing with django-project

### ISSUE THREAD

**Issue by cesar-benjamin** [GeoNode/geonode#11922]
## Expected Behavior

Update admin password on install from tasks.py when installing with django-project

## Actual Behavior

updateadmin() throws an "/bin/bash: line 1: django-admin.py: command not found" message.

## Steps to Reproduce the Problem

  1. Install master using geonode-project
  2. view the logs
  3. look at ***********************update admin details************************** in logs

## Specifications

  - GeoNode version: master 
  - Installation type (vanilla, geonode-project): geonode-project
  - Installation method (manual, docker): docker
  - Platform: ubuntu 22.04
  - Additional details: i think can be fixed changing line 426 in updateadmin from tasks.py changing f"django-admin loaddata /tmp/django_admin_docker.json to f"python manage.py loaddata /tmp/django_admin_docker.json

**Comment by sistein** [comment]
I think I'm facing the same issue. 

`django4ageing_geonode       | ***********************update admin details**************************
django4ageing_geonode       | /bin/bash: line 1: django-admin.py: command not found
dependency failed to start: container django4ageing_geonode is unhealthy
`

**Comment by cesar-benjamin** [comment]
I have a lot of open issues, and contribs but they don't care. But they took my work as own on the upgrade to django 4.2 without credit on me

**Comment by cesar-benjamin** [comment]
Then I decide to stop working on fixing things with this approach from them, so I guess I will build my own SDI without the lot of arcane things and obsolete code. For example, in user profile there is a few counters per dataset which don't work because uses AngularJS... AngularJS! At this time! And of course mixed with react and jquery... Is a shame

**Comment by ridoo** [comment]
> I have a lot of open issues, and contribs but they don't care. But they took my work as own on the upgrade to django 4.2 without credit on me

@cesar-benjamin I saw your comments on upgrading Django version, but I cannot judge on this because I was not involved here. However, I cannot find any PR [which was worth](https://github.com/GeoNode/geonode/pulls?q=is%3Apr+author%3Acesar-benjamin) to merge and give credit for. Maybe I miss some locations where to look at. There are so many, as GeoNode is big and there are lots of components involved. If so, please point me on it.

> Then I decide to stop working on fixing things with this approach from them, so I guess I will build my own SDI without the lot of arcane things and obsolete code. For example, in user profile there is a few counters per dataset which don't work because uses AngularJS... AngularJS! At this time! And of course mixed with react and jquery... Is a shame

Good luck on building your own. Curious to see, how it will look in ten years from now.

**Comment by giohappy** [comment]
@cesar-benjamin I'm sorry to hear such feedback. I don't want to argue against your feelings but let me answer in a few points:

> but they don't care. But they took my work as own on the upgrade to django 4.2 without credit on me

I'm not sure what  "they" is in your perception of this project. Some of us work in a company (GeoSolutions) that uses and contributes to GeoNode. We're not the owners of GeoNode. It is and will remain a community project.
We do our best to maintain it, being the principal contributors at the moment. However, this project started way before us. Many hands worked on it, and many parts have been written by different people and groups, with multiple programming styles, mindsets, etc. There are a lot of good things but also a significant tech depth in some parts. 
During the upgrade to GeoNode 4.0, which was financed by a project from my company. We did our best to refactor some parts, but not everything could be improved, otherwise a full rewrite was needed.

Let me say that if you can take advantage of the good parts of GeoNode it is thanks to many people who worked on it, spent time and money on it, and decided to keep it open for everyone. 

On our side, we don't have any reason to block things and put up fences. We spend (**free**) time reviewing proposals and PRs. and merging or taking what looks good, considering also the many workflows that depend on the code.
For example, your proposal to change the way requirements were written was breaking some CD workflows. We have spent some time figuring out why your solution wasn't working but, I repeat, we were doing it for free and we couldn't afford digging too much. 
Should we accept it blindly and let everyone adapt to such a breaking change?

Regarding the upgrade to Django 4. We were already working on it, then we saw your PRs (many of which were closed by you in a quite confusing flow, I have to say). We took the good parts and integrated them into the work we were going to push.
There are no credits to GeoSolutions for this work, but if you want we can put your name in the changelog, no problem.

I don't want to start a flame for no reason, and I apologize if my answer could sound a bit harsh, but please take into account the whole thing before judging. We think your contributions are valuable and I hope you will continue to push them. I cannot guarantee we will be able to respond or review promptly. Nobody pay us for it :)

**Comment by mattiagiupponi** [comment]
> I have a lot of open issues, and contribs but they don't care. But they took my work as own on the upgrade to django 4.2 without credit on me

I give you the credit in the GNIP created for the upgrade of Django to 4.2 as you can see here: https://github.com/GeoNode/geonode/issues/11821
![image](https://github.com/GeoNode/geonode/assets/51856725/0926dc18-86cf-4713-9549-c5b069a0c06a)

![image](https://github.com/GeoNode/geonode/assets/51856725/4ea5e3fa-4071-40ec-b55c-cedfb701389b)

Just a couple of notes:
- Your initial upgrade was done on the branch 4.1.x which was not mantained anymore.
- I used the work you have done to upgrade Django on master branch as was described in more than one occasion 
- You work has been merged more than once both in `geonode` and in the `geonode-importer`
- If a PR is in draft state, to me is not a PR ready for any review example https://github.com/GeoNode/geonode/pull/11921
- About this Issue i just see a PR from you that was open and closed by you without any review https://github.com/GeoNode/geonode/pull/11923

**Comment by giohappy** [comment]
And a final note @cesar-benjamin about this issue: I don't see it connected to any PR.

**Comment by sistein** [comment]
Coming back to my issue: 

I have a fresh installation of geonode using docker. First startup fails because of the missing django-admin.py 

`django4ageing_geonode  | ***********************update admin details**************************
django4ageing_geonode  | /bin/bash: line 1: django-admin.py: command not found
django4ageing_geonode exited with code 0`

I tried setting it up multiple times. The only difference to the how-to is using debian and docker compose instead of docker-compose. 

I just cannot really believe that I'm the only person with this issue.

**Comment by mattiagiupponi** [comment]
Hi @sistein 

> I just cannot really believe that I'm the only person with this issue.

I made quite few project with GeoNode and i never saw this problem, but let's try in step:

1) GeoNode version
2) Docker version

Then few action to try:

- Did you try to rebuild the project ia docker with the `--no-cache` command? like `docker compose build --no-cache`
- Please try to do a `docker compose down --volumes` to clean the whole project and removing all the volumes along with it
- Once the build is done please try with `docker compose up -d --force-recreate`

Let me know if some of this steps helped

**Comment by cesar-benjamin** [comment]
> Coming back to my issue:
> 
> I have a fresh installation of geonode using docker. First startup fails because of the missing django-admin.py
> 
> `django4ageing_geonode | ***********************update admin details************************** django4ageing_geonode | /bin/bash: line 1: django-admin.py: command not found django4ageing_geonode exited with code 0`
> 
> I tried setting it up multiple times. The only difference to the how-to is using debian and docker compose instead of docker-compose.
> 
> I just cannot really believe that I'm the only person with this issue.

i had pointed this issue too, but obviously i was ignored, but i can tell you how to fix it: you have to edit the tasks.py file and find and replace all **django-admin.py** entries to **python manage.py** and then all will go fine.

**Comment by cesarbenjamindotnet** [comment]
> We took the good parts and integrated them into the work we were going to push.
> There are no credits to GeoSolutions for this work, but if you want we can put your name in the changelog, no problem.

@giohappy: Yes please. i think a lot of work made by me make possible this upgrade because i did a deep research abput the broken libs and so on, i know that not all did work fine but was a huge work that make it possible the upgrade. So please mantion that in the changelog, i don't saw it yet.

Thanks.

**Comment by kikislater** [comment]
@cesarbenjamindotnet : It seems most of your PRs are being closed because your name isn’t listed in the .clabot file for this project: https://github.com/GeoNode/geonode/blob/master/.clabot.

I’m not affiliated with GeoSolutions, and while I agree it’s not the most convenient system, the requirement is clearly mentioned in the contribution guidelines here: https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md.

To have your PRs accepted and avoid them being rejected by the CLA bot, you’ll need to sign the CLA.
